integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -668,6 +691,81 @@ public String getFileSystemName() {
     return mFsName;
   }
 
+  @Override
+  public void umount() {
+    // Release operation is async, we will try our best efforts to
+    // close all opened file in/out stream before umounting the fuse
+    if (!mCreateFileEntries.isEmpty() || !mOpenFileEntries.isEmpty()) {
+      LOG.info("Unmounting {}. Waiting for all in progress file read/write to finish", mMountPoint);
+      try {
+        CommonUtils.waitFor("all in progress file read/write to finish",
+            () -> mCreateFileEntries.isEmpty() && mOpenFileEntries.isEmpty(),
+                WaitForOptions.defaults().setTimeoutMs(mMaxUmountWaitTime));
+      } catch (InterruptedException e) {
+        LOG.error("Unmount {} interrupted", mMountPoint);
+        Thread.currentThread().interrupt();
+      } catch (TimeoutException e) {
+        LOG.error("Timeout when waiting all in progress file read/write to finish "
+            + "when unmounting {}. {} fileInStream remain unclosed. "
+            + "{} fileOutStream remain unclosed.",
+            mMountPoint, mOpenFileEntries.size(), mCreateFileEntries.size());
+      }
+    }
+
+    // Waiting for in progress async release to finish
+    if (!mReleasingReadEntries.isEmpty() || !mReleasingWriteEntries.isEmpty()) {
+      LOG.info("Unmounting {}. Waiting for all in progress file read/write closing to finish",
+          mMountPoint);
+      try {
+        CommonUtils.waitFor("all in progress file read/write closing to finish",
+            () -> mReleasingReadEntries.isEmpty() && mReleasingWriteEntries.isEmpty(),
+            WaitForOptions.defaults().setTimeoutMs(mMaxUmountWaitTime));
+      } catch (InterruptedException e) {
+        LOG.error("Unmount {} interrupted", mMountPoint);
+        Thread.currentThread().interrupt();
+      } catch (TimeoutException e) {
+        LOG.error("Timeout when waiting in progress file read/write closing to finish "
+            + "when unmounting {}. {} fileInStream and {} fileOutStream "
+            + "are still in closing process.",
+            mMountPoint, mReleasingReadEntries.size(), mReleasingWriteEntries.size());
+      }
+    }
+
+    // Force close all opened stream
```

Comment:
I don't think we shall force close all opened streams. Might be better to 
`throw new FuseException("Device or resource busy")` as pointed out by @Binyang2014

Reply:
We throw FuseException to not allow umount?
Not sure if it's correct. I think we should always allow umount, otherwise the fuse mount state may be strange or we cannot kill the process?

Reply:
It's possible that some open()/create() may not be properly release() as client operation crash?
In that case, i would not want to prevent umount from happening.
As for whether we want to force close opened streams, we can discuss that part.

Reply:
@LuQQiu  in reality, umount a busy mount point is expect to fail there. This is often useful to help avoid data loss by disallowing unmouting a filesystem when it is being used.

Reply:
maybe leave a TODO here to implement umount with force ?

Reply:
Sounds good. I will do more testing for it tomorrow morning. Feeling a little unsafe to throw the exception

Reply:
Actually, I think OS don't let we umount a busy device. Throw exception is good for me. Thanks